services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.controller.ts

```diff
@@ -1,128 +1,335 @@
-  }
+    private readonly programFinancialServiceProviderConfigurationsService: ProgramFinancialServiceProviderConfigurationsService,
+  ) {}
 
   @AuthenticatedUser({ isAdmin: true })
   @ApiOperation({
-    summary: 'Get all programFspConfigurationEntity for a specific program',
+    summary: 'Get all Financial Service Provider Configurations for a Program.',
   })
   @ApiParam({ name: 'programId', required: true, type: 'integer' })
   @ApiResponse({
     status: HttpStatus.OK,
-    description: 'Return programFspConfigurationEntity by program id.',
+    description:
+      'Return Financial Service Provider Configurations by Program Id.',
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: 'Program does not exist',
   })
-  @Get(':programId/fsp-configuration')
+  @Get(':programId/financial-service-provider-configurations')
   public async findByProgramId(
     @Param('programId', ParseIntPipe)
     programId: number,
-  ): Promise<ProgramFinancialServiceProviderConfigurationEntity[]> {
-    return this.programFspConfigurationService.findByProgramId(programId);
+  ): Promise<ProgramFinancialServiceProviderConfigurationResponseDto[]> {
+    return this.programFinancialServiceProviderConfigurationsService.findByProgramId(
+      programId,
+    );
   }
 
   @AuthenticatedUser({ isAdmin: true })
   @ApiOperation({
-    summary: 'Create ProgramFspConfigurationEntity for a program',
+    summary:
+      'Create a Financial Service Provider Configuration for a Program. Financial Service Provider Configuration properties',
   })
   @ApiParam({ name: 'programId', required: true, type: 'integer' })
   @ApiResponse({
-    status: 201,
+    status: HttpStatus.CREATED,
     description:
-      'The programFspConfigurationEntity has been successfully created.',
+      'The Financial Service Provider Configuration has been successfully created.',
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: 'Program does not exist',
   })
-  @ApiResponse({ status: HttpStatus.BAD_REQUEST, description: 'Bad request.' })
-  @Post(':programId/fsp-configuration')
+  @ApiResponse({
+    status: HttpStatus.BAD_REQUEST,
+    description: 'Bad request. Body or params are malformed',
+  })
+  @ApiResponse({
+    status: HttpStatus.CONFLICT,
+    description:
+      'Program Financial Service Provider Configuration with same name already exists',
+  })
+  @Post(':programId/financial-service-provider-configurations')
   public async create(
-    @Body() programFspConfigurationData: CreateProgramFspConfigurationDto,
+    @Body()
+    programFspConfigurationData: CreateProgramFinancialServiceProviderConfigurationDto,
     @Param('programId', ParseIntPipe)
     programId: number,
-  ): Promise<number> {
-    return await this.programFspConfigurationService.create(
+  ): Promise<ProgramFinancialServiceProviderConfigurationResponseDto> {
+    return await this.programFinancialServiceProviderConfigurationsService.validateAndCreate(
```

Comment:
rename to .create(): consistent with other functions in the service and other services. The .update function also does validation, so no reason to call this function validateAndCreate.